### PR TITLE
Bug 1217381 - Go back to default collection settings on logout. r=michielbdejong, yifan

### PR DESCRIPTION
**Pull request by ferjm**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=0a39322934b9806f115dd7d39c15160d2633c830

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=78f4ebd46a3ac13ab972e89bce682fcbce618a6d

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=b66ab8ee0c18fb06672c1702b2195b1098977f63

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=e3fbb549065b60b355a2287098e6be1cfb4d499d

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=ec6d03b4dee1723ff93a4d51bf8fcb343426137a

File: tv_apps/browser/js/sync/settings.js [hunk 1]
```diff
@@ -15,7 +15,7 @@
 'use strict';
 
 (function(exports) {
-  const DEBUG = false;
+  const DEBUG = true;
```
Comment (begeeben) [inline tv_apps/browser/js/sync/settings.js]: I guess we should change it back to false before landing?

File: tv_apps/browser/js/sync/settings.js [hunk 2]
```diff
@@ -92,6 +92,8 @@
 
       this.showScreen(DISABLED);
 
+      this.showScreen(DISABLED);
```
Comment (begeeben) [inline tv_apps/browser/js/sync/settings.js]: Duplicates.

File: tv_apps/smart-system/js/sync_manager.js [hunk 3]
```diff
@@ -549,6 +633,14 @@
         });
         promises.push(promise);
       });
+
+      SETTINGS.forEach(setting => {
+        var promise = new Promise(resolve => {
+          asyncStorage.removeItem(setting, resolve);
+          promises.push(promise);
+        });
+      });
```
Comment (begeeben) [inline tv_apps/smart-system/js/sync_manager.js]: If a user logs out and restarts b2g, will her saved settings get deleted?

File: apps/system/js/sync_manager.js [hunk 4]
```diff
@@ -252,7 +252,12 @@
     _handle_onsyncdisabling: function() {
       this.debug('onsyncdisabling observed');
       this.updateState();
-      Service.request('SyncStateMachine:success');
+
+      // Go back to default settings, so new users won't inherit old users
```
Comment (michielbdejong) [inline apps/system/js/sync_manager.js]: Nit: s/old users preferences/old users' preferences

File: apps/system/js/sync_manager.js [hunk 5]
```diff
@@ -313,6 +318,11 @@
       }).then(() => {
         return this.getAccount();
       }).then(() => {
+        // We save default settings so we can revert user changes to the
+        // default values when the user logs out from Sync. So new users don't
+        // inherit old users preferences.
```
Comment (michielbdejong) [inline apps/system/js/sync_manager.js]: Nit: s/old users preferences/old users' preferences

File: tv_apps/browser/js/sync/settings.js [hunk 6]
```diff
@@ -261,6 +295,13 @@
         navigator.mozId.watch({
           wantIssuer: 'firefox-accounts',
           onlogin: () => {
+            // This callback is fired everytime a new account logs in,
```
Comment (michielbdejong) [inline tv_apps/browser/js/sync/settings.js]: Nit: s/everytime/every time

File: tv_apps/browser/js/sync/settings.js [hunk 7]
```diff
@@ -261,6 +295,13 @@
...
+            // but we only want to enable the history setting when this
+            // callback is called because the refresh auth flow succeeded.
```
Comment (michielbdejong) [inline tv_apps/browser/js/sync/settings.js]: This code seems to be present in `onhistorychecked`, but not in `onbookmarkschecked` - why this asymmetry?

Comment (michielbdejong) [inline tv_apps/browser/js/sync/settings.js]: Also, from the code it seems `onhistorychecked` is called both when `this.elements.collectionHistory.checked` changes to `true`, and when it changes to `false`. But it seems the checkbox is always unchecked after a `navigator.mozId` error, here: https://github.com/mozilla-b2g/gaia/pull/32779/files#diff-50ac43c342df1ca3d63486ce5b2a1057R313, isn't that also an asymmetry?

Comment (ferjm) [inline tv_apps/browser/js/sync/settings.js]: Regarding the asymmetry: https://bugzilla.mozilla.org/show_bug.cgi?id=1220541

Comment (ferjm) [inline tv_apps/browser/js/sync/settings.js]: We only call the refresh auth flow when the checkbox changes to true and we need to uncheck it if the refresh auth fails (onerror and/or oncancel callbacks).

Comment (michielbdejong) [inline tv_apps/browser/js/sync/settings.js]: Ah, ok.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=0c66091974e2f67800f487a7fe88f0668061d064

**Pull request merged**